### DISCUSSION TITLE
iam_service.json not found

### DISCUSSION THREAD

**Discussion by bytewiz** [googleapis/nodejs-pubsub#1315] [Q&A]
## ERROR 
I get the following error when I try to use the pubsub client. In my cloud function environment calling pubsub nodejs client. 

`Warning: iam_service.json not found in any of the include paths`

#### Environment details

  - OS: `Cloud functions`
  - Node.js version: `12.20`
  - npm version:  `yarn 1.18`
  - `@google-cloud/pubsub` version: `2.10`

#### Steps to reproduce

Trying to instantiate a pubsub client 
```typescript
/**
*  This service account has full access to pubsub on this project.
*  I logged all credentials as well and they are correct.
*/
const sa = functions.config().pubsub.key;

const pubsubClient = new PubSub({
    projectId: sa.project_id,
    credentials: sa, // this is the service account json file (also tried only as { client_email, private_key } same error
});
```
Call function where client is used:

```typescript 
export const publishToTopic = async (details: { topic: string; body: any }) => {
    try {
        const data = JSON.stringify(details.body);
        const dataBuffer = Buffer.from(data);

        await pubsubClient.topic(details.topic).publish(dataBuffer);

        return;
    } catch (error) {
        console.error(JSON.stringify(error));
    }
};
```
  
Get error `Warning: iam_service.json not found in any of the include paths`

Any help would be appreciated! Due to how cloud functions work I can't seem to use the option of passing a `keyFilename` path 

Note I did try following this: https://github.com/googleapis/nodejs-pubsub/issues/1000
but it still throws `IAM error`

**Comment by feywind** [comment]
@bytewiz That sounds annoying. Let me see if I can find someone who knows a bit more about Cloud Functions (@bcoe ?). If that doesn't turn anything up, I'll be able to look at this next week.

**Comment by bytewiz** [comment]
> @bytewiz That sounds annoying. Let me see if I can find someone who knows a bit more about Cloud Functions (@bcoe ?). If that doesn't turn anything up, I'll be able to look at this next week.

Thanks that would be great! Would really love to use it so that would be helpful 🙏

**Comment by bytewiz** [comment]
@feywind nothing after 2 weeks? 😞  would be nice to know something about this, as we plan to use pubsub quite a bit with cloud functions

**Comment by bytewiz** [comment]
@alexander-fenster Are you able to help out here? :) Would be much appreciated
